### DISCUSSION TITLE
The problem of running tauri on centos8

### DISCUSSION THREAD

**Discussion by hanxiaohuivc** [tauri-apps/tauri#2463] [Q&A] [answered]
The problem of running tauri on centos8
Hello everyone. I deployed tauri on centos8, but the following error occurred when using "yarn tauri dev",

error: failed to run custom build command for `glib-sys v0.10.1`
--- stderr
   `"pkg-config" "--libs" "--cflags" "glib-2.0" "glib-2.0 >= 2.44"` did not exit successfully: exit status: 1
   --- stderr
   Package glib-2.0 was not found in the pkg-config search path.
   Perhaps you should add the directory containing `glib-2.0.pc'
   to the PKG_CONFIG_PATH environment variable
   Package'glib-2.0', required by'virtual:world', not found
   Package'glib-2.0', required by'virtual:world', not found

Is the cause of this error missing gtk on my system? Do you have an effective way to solve this problem?
Thank you.

**Comment by amrbashir** [comment]
did you install the system dependencies listed here https://tauri.studio/en/docs/get-started/setup-linux#1-system-dependencies ?

**Reply by hanxiaohuivc** [reply]
I have successfully installed them.

**Reply by amrbashir** [reply]
you also need to install `gtk` on the machines that will run the app, which you said you didn't install.

**Reply by hanxiaohuivc** [reply]
Yes, I did not install gtk.

**Reply by sudo-vaibhav** [reply]
this link doesnt work now

**Reply by FabianLars** [reply]
https://v1.tauri.app/v1/guides/getting-started/prerequisites#setting-up-linux

**Comment by hanxiaohuivc** [comment, marked as answer]
Finally, I successfully installed tauri on centos7, provided that glibc is upgraded to 2.3. It is estimated that it should work on centos8.

**Comment by Myrmod** [comment]
As reference for other people that might land on this issue. I had to install gtk manually and restart my terminals/vscode

```
sudo apt-get install libgtk-3-dev
```

Afterwards it worked.

**Comment by kwiecien** [comment]
In my case, the same error was caused by build conflict with `homebrew` (see  https://tauri.app/v1/guides/faq/#build-conflict-with-homebrew-on-linux)
I solved it by uninstalling the `homebrew` but another solution (according to the above page) is to change the environment variable `PKG_CONFIG_PATH`
